js/common/modules/@arangodb/test-helper-common.js

```diff
@@ -28,12 +28,45 @@
 // / @author Copyright 2011-2012, triAGENS GmbH, Cologne, Germany
 // //////////////////////////////////////////////////////////////////////////////
 
+const arangodb = require('@arangodb');
+const db = arangodb.db;
 let internal = require('internal'); // OK: processCsvFile
 const request = require('@arangodb/request');
 const fs = require('fs');
 
 let instanceInfo = null;
 
+exports.transactionFailure = function (trx, errorCode, errorMessage, abortArangoshOnly, crashOnSuccess) {
+  try {
+    db._executeTransaction(trx);
+  } catch (ex) {
+    if ((!abortArangoshOnly || global.arango) &&  // only on arangosh...
+        (ex instanceof arangodb.ArangoError) && // only regular arango errors has the subsequent:
+        (ex.errorNum === errorCode) && // check for right error code
+        (!errorMessage || (ex.message === `Error: ${errorMessage}`))) { // optional errorMessage
+      if (crashOnSuccess) {
+        internal.debugTerminate('crashing server');
+      }
+      return 0;
+    }
+    console.log(ex);
+  }
+  return 1;
+};
+
+exports.truncateFailure = function (collection) {
+  try {
+    collection.truncate();
+    return 1;
+  } catch (ex) {
+    if (!ex instanceof arangodb.ArangoError ||
+        ex.errorNum !== internal.errors.ERROR_DEBUG.code) {
+      throw ex;
+    }
+  }
+  internal.debugTerminate('crashing server');
+};
+
```

Comment:
Maybe:
`(func, errorCode, errorMessage = null, abortArangoshOnly = false, crashOnSuccess = true)`?

Reply:
I don't like the idea of a single function because of you cannot read the test code well anymore without having to read the lib code. This clearly voids KISS principle.

Reply:
Why? It's just called "func", other is related to CI.

func is lambda so it more easy to understood then some special functions for colelction